feature-engineering/categorical_imputer.md

```diff
@@ -0,0 +1,153 @@
+Impute missing categorical values using reference features.
+
+The imputer takes as an input a column which contains categorical values, and may contain
+some None values. Its secondary input is a set of reference feature columns. During the 
+Fit phase, the imputer learns the relationships between the values of the column to be
```

Comment:
Are these phases part of the user interaction surface of the imputer (does the user have fine-grained control over phases)? If not, can we abstract this concept away from the user and just say "The imputer learns the relationships between the values of the column to be imputed and the reference feature columns, then applies that learned relationship to the missing values to fill them with categorical values."